ramalama/daemon/daemon.py

```
@@ -115,19 +128,28 @@ def shutdown(self):
-    logger.debug(f"Starting Ramalama daemon on {host}:{port}...")
-    with RamalamaServer(host, port, model_store_path, timedelta(seconds=10)) as httpd:
-        with ShutdownHandler(httpd):
-            server_thread = threading.Thread(target=httpd.serve_forever, daemon=True)
+    host_str = f"[{host}]" if ":" in host else host
+    logger.debug(f"Starting Ramalama daemon on {host_str}:{port}...")
+    try:
+        server = RamalamaServer(host, port, model_store_path, timedelta(seconds=10))
+    except OSError:
+        if host != "::":
+            raise
+        host = "0.0.0.0"
```

Ruff (0.15.10)

[error] Possible binding to all interfaces

(S104)